src/app/advanced-examples/minesweeper/components/controls.ts

```diff
@@ -0,0 +1,47 @@
+import { Div, Button, classes, event } from "rxfm";
+import { Observable, combineLatest, of, timer } from "rxjs";
+import { switchMap, filter, map, scan, startWith } from "rxjs/operators";
+import { GameStage, CellAction } from "../types";
+
+export const Controls = (
+  startTime: Observable<number | undefined>,
+  endTime: Observable<number | undefined>,
+  gameStage: Observable<GameStage>,
+  dispatch: (action: CellAction) => void,
+) => {
+  const GameTime = combineLatest([startTime, endTime]).pipe(
+    switchMap(([start, end]) => {
+      if (!start) return of(0);
+      else if (!end) return timer(0, 1000);
+      return of(Math.round((end - start) / 1000));
+    }),
+    switchMap(time => Div('Time: ', time, 's')),
```

Comment:
Need distinct until changed here or can't use switchMap. Same goes for the below components.

Reply:
fixed